File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/DataSourceMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.model.DataSource
import com.rookmotion.rook_sdk_health_connect.proto.DataSourceProto

fun DataSource.toDataSourceProto(): DataSourceProto {
    val dataSourceProtoBuilder = DataSourceProto.newBuilder()
        .setName(name)
        .setDescription(description)
        .setImage(image)
        .setConnected(connected)

    if (authorizationUrl != null) {
        dataSourceProtoBuilder.setAuthorizationUrl(authorizationUrl)
        dataSourceProtoBuilder.setAuthorizationUrlIsNull(false)
    } else {
        dataSourceProtoBuilder.setAuthorizationUrlIsNull(true)
    }

    return dataSourceProtoBuilder.build()
}

File: packages/rook_sdk_health_connect/lib/src/data/mapper/data_source_mappers.dart
import 'package:rook_sdk_core/rook_sdk_core.dart';
import 'package:rook_sdk_health_connect/src/data/proto/protos.pb.dart';

extension DataSourceMappers on DataSourceProto {
  DataSource toDomain() {
    return DataSource(
      name,
      description,
      imageUrl,
      imageUrl,
      connected,
      authorizationUrlIsNull ? null : authorizationUrl,
    );
  }
}
